packages/tests/server/websockets.test.ts

```diff
@@ -1202,3 +1206,56 @@ describe('lazy mode', () => {
     await ctx.close();
   });
 });
+
+describe('keep alive', () => {
+  const keepAliveConfig = {
+    pingMs: 2000,
+    pongWaitMs: 5000,
+  };
+  test('pong is received', async () => {
+    const ctx = factory({
+      wssServer: {
+        keepAlive: {
+          enabled: true,
+          ...keepAliveConfig,
+        },
+      },
+    });
+    const pongMock = vi.fn();
+    const { wsClient, wss } = ctx;
+    wss.on('connection', (ws) => {
+      ws.on('pong', pongMock);
+    });
+    {
+      expect(wsClient.connection).not.toBe(null);
+      await waitMs(3500);
```

Comment:
Please use [`useFakeTimers`](https://vitest.dev/guide/mocking.html#example) instead of real waiting so the test is snappy